rx/core/operators/distinct.py

```diff
@@ -22,9 +22,13 @@ def push(self, value):
             self.set.append(value)
         return ret_value
 
+    def flush(self):
+        self.set = []
```

Comment:
I believe we need some synchronization primitive here, either in `flush`/`push` methods or at caller scope by using the built-in `Observable.lock` (might be tricky).  

Given the following sample, values will be checked or appended to `HashSet.set` on thread A, while the set will be wipe out on thread B. 

```python
import rx
from rx import operators as ops

flushes = rx.interval(1.0)

rx.interval(0.4).pipe(
    ops.distinct(flushes=flushes),
    ).subscribe(print)
```

Reply:
Thanks. I'll take a look into that

Reply:
Thanks for the update. I find the usage of ExitStack quite verbose for such a simple usage. using nullconext may be more concise and expressive. For example:

    def flush(self):
        with self.lock or contextlib.nullcontext():
            self.set = []

I let you @jcafhe see what option you prefer.

Reply:
Ok, I'm cool with that

Reply:
I agree its a cleaner approach

Reply:
@MainRo nullcontext seems to be new in 3.7 and breaks the 3.6 build

Reply:
@MainRo I extracted out nullcontext for 3.6 compatibility. There's almost no code to it, so that might be OK. What do you think?

Reply:
@jcafhe thoughts?

Reply:
That looks fine to me, thank you :+1

Reply:
excellent!